### PR TITLE
neovim-gtk: init at 1.0.4

### PR DESCRIPTION
**Pull request by Aleksanaa**
###### Description of changes

https://github.com/Lyude/neovim-gtk. This is a new branch of neovim-gtk with gtk4 supported. I've used it for a while and eveything works.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/neovim/neovim-gtk.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+, vte-gtk4
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "neovim-gtk";
+  version = "1.0.4";
+
+  src = fetchFromGitHub {
+    owner = "Lyude";
+    repo = "${pname}";
```
Comment (bobby285271) [inline pkgs/applications/editors/neovim/neovim-gtk.nix:19]:
```suggestion
    repo = pname;
```

File: pkgs/applications/editors/neovim/neovim-gtk.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    rev = "v${version}";
+    hash = "sha256-inva7pYwOw3bXvFeKZ4aKSQ65iCat5HxM+NME8jN4/I=";
+  };
+
+  cargoHash = "sha256-9eZwCOP4xQtFOieqVRBAdXZrXmzdnae6PexGJ/eCyYc=";
+
+  nativeBuildInputs = [ wrapGAppsHook pkg-config glib gdk-pixbuf ];
```
Comment (bobby285271) [inline pkgs/applications/editors/neovim/neovim-gtk.nix:26]: For GTK4 programs:

```suggestion
  nativeBuildInputs = [ wrapGAppsHook4 pkg-config glib gdk-pixbuf ];
```

File: pkgs/applications/editors/neovim/neovim-gtk.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  buildInputs = [ gtk4 pango vte-gtk4 ];
+
+  postInstall = ''
+    cd $src
+    make DESTDIR=/ PREFIX=$out install-resources
+  '';
+
+  meta = with lib; {
+    description = "Gtk ui for neovim. ";
```
Comment (bobby285271) [inline pkgs/applications/editors/neovim/neovim-gtk.nix:36]: [CONTRIBUTING.md says avoid trailing period](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes)

```suggestion
    description = "Gtk ui for neovim";
```

File: pkgs/applications/editors/neovim/neovim-gtk.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    homepage = "https://github.com/Lyude/${pname}";
```
Comment (bobby285271) [inline pkgs/applications/editors/neovim/neovim-gtk.nix:37]: We want the link clickable.

```suggestion
    homepage = "https://github.com/Lyude/neovim-gtk";
```

**Comment by Aleksanaa** [comment]
Okay done

**Comment by Aleksanaa** [comment]
Anyone please review this PR. It contains maintainer metadata and blocks my other contributions.

File: pkgs/applications/editors/neovim/neovim-gtk.nix [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  src = fetchFromGitHub {
+    owner = "Lyude";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-inva7pYwOw3bXvFeKZ4aKSQ65iCat5HxM+NME8jN4/I=";
+  };
+
+  cargoHash = "sha256-9eZwCOP4xQtFOieqVRBAdXZrXmzdnae6PexGJ/eCyYc=";
+
+  nativeBuildInputs = [ wrapGAppsHook4 pkg-config glib gdk-pixbuf ];
```
Comment (figsoda) [inline pkgs/applications/editors/neovim/neovim-gtk.nix:26]: I don't think it's using anything from `glib.bin`, and `gdk-pixbuf` should probably be in `buildInputs` instead

```suggestion
  nativeBuildInputs = [ wrapGAppsHook4 pkg-config ];
```

File: pkgs/applications/editors/neovim/neovim-gtk.nix [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+  buildInputs = [ gtk4 pango vte-gtk4 ];
```
Comment (figsoda) [inline pkgs/applications/editors/neovim/neovim-gtk.nix:28]:
```suggestion
  buildInputs = [ gdk-pixbuf gtk4 pango vte-gtk4 ];
```

File: pkgs/applications/editors/neovim/neovim-gtk.nix [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+  postInstall = ''
+    cd $src
+    make DESTDIR=/ PREFIX=$out install-resources
```
Comment (figsoda) [inline pkgs/applications/editors/neovim/neovim-gtk.nix:32]:
```suggestion
    make PREFIX=$out install-resources
```

File: pkgs/applications/editors/neovim/neovim-gtk.nix [hunk 8]
```diff
@@ -0,0 +1,41 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, wrapGAppsHook4
+, pkg-config
+, gtk4
+, glib
+, pango
+, gdk-pixbuf
+, vte-gtk4
```
Comment (figsoda) [inline pkgs/applications/editors/neovim/neovim-gtk.nix:10]:
```suggestion
, rustPlatform
, fetchFromGitHub
, wrapGAppsHook4
, pkg-config
, gdk-pixbuf
, gtk4
, pango
, vte-gtk4
```

File: pkgs/applications/editors/neovim/neovim-gtk.nix [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  postInstall = ''
+    cd $src
+    make DESTDIR=/ PREFIX=$out install-resources
+  '';
+
+  meta = with lib; {
+    description = "Gtk ui for neovim";
+    homepage = "https://github.com/Lyude/neovim-gtk";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ aleksana ];
```
Comment (figsoda) [inline pkgs/applications/editors/neovim/neovim-gtk.nix:39]:
```suggestion
    maintainers = with maintainers; [ aleksana ];
    mainProgram = "nvim-gtk";
```

File: pkgs/applications/editors/neovim/neovim-gtk.nix [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  cargoHash = "sha256-9eZwCOP4xQtFOieqVRBAdXZrXmzdnae6PexGJ/eCyYc=";
+
+  nativeBuildInputs = [ wrapGAppsHook4 pkg-config ];
+
+  buildInputs = [ gdk-pixbuf gtk4 pango vte-gtk4 ];
+
+  postInstall = ''
+    make DESTDIR=/ PREFIX=$out install-resources
```
Comment (figsoda) [inline pkgs/applications/editors/neovim/neovim-gtk.nix:30]:
```suggestion
    make PREFIX=$out install-resources
```

**Review by figsoda — APPROVED** [review]

**Pull request merged**